api/api.hh

```diff
@@ -237,6 +237,66 @@ public:
     operator T() const { return value; }
 };
 
+using mandatory = bool_class<struct mandatory_tag>;
```

Comment:
I understand why you took this approach, but at the end of the day, url parameters are string (or list of string)
I would have take one of two approaches, either continue treat them as string, we have enought string manipulation functions out there.

Or two, use the swagger (just like we do for response) and validate the request, if you want to feel posh,  we can create a struct per request.

Reply:
I think the ideal long-term solution should use the json model we define and generate a completely verifiable representation that the seastar httpd layer can use.

Reply:
exacltly, that's why i don't like the approach of adding this parameter handling in Scylla, especially when it's only be used here.

Reply:
The mid-term plan is to use it extensively, until there's complete validation at the seastar layer, integrated with scylla code generation for the api interface, as discussed above.